MaterialLib/SolidModels/Ehlers.h

```diff
@@ -0,0 +1,213 @@
+        /// see Ehler's paper "A single-surface yield function for geomaterials"
+        /// for more details.
+        MaterialProperties(P const& G_, P const& K_, P const& alpha_,
+                           P const& beta_, P const& gamma_, P const& delta_,
+                           P const& epsilon_, P const& m_, P const& alpha_p_,
+                           P const& beta_p_, P const& gamma_p_,
+                           P const& delta_p_, P const& epsilon_p_,
+                           P const& m_p_, P const& kappa_,
+                           P const& hardening_coefficient_)
+            : G(G_),
+              K(K_),
+              alpha(alpha_),
+              beta(beta_),
+              gamma(gamma_),
+              delta(delta_),
+              epsilon(epsilon_),
+              m(m_),
+              alpha_p(alpha_p_),
+              beta_p(beta_p_),
+              gamma_p(gamma_p_),
+              delta_p(delta_p_),
+              epsilon_p(epsilon_p_),
+              m_p(m_p_),
+              kappa(kappa_),
+              hardening_coefficient(hardening_coefficient_)
+        {
+        }
+        // basic material parameters
+        P const& G;  ///< shear modulus
+        P const& K;  ///< bulk modulus
+
+        P const& alpha;
+        P const& beta;
+        P const& gamma;
+        P const& delta;
+        P const& epsilon;
+        P const& m;
+
+        P const& alpha_p;
+        P const& beta_p;
+        P const& gamma_p;
+        P const& delta_p;
+        P const& epsilon_p;
+        P const& m_p;
+
+        P const& kappa;
+        P const& hardening_coefficient;
+        // Drucker-Prager: Import kappa and beta in terms of Drucker-Prager
+        // criterion solution dependent values
+        double k;
+
+        void calculateIsotropicHardening(double const t,
+                                         ProcessLib::SpatialPosition const& x,
+                                         const double e_pv_curr);
+    };
+
+    struct MaterialStateVariables
+        : public MechanicsBase<DisplacementDim>::MaterialStateVariables
+    {
+        void loadState()
+        {
+            eps_p_D = eps_p_D_prev;
+            eps_p_V = eps_p_V_prev;
+            eps_p_eff = eps_p_eff_prev;
+            lambda = 0;
+        }
+
+        void pushBackState() override
+        {
+            eps_p_D_prev = eps_p_D;
+            eps_p_V_prev = eps_p_V;
+            eps_p_eff_prev = eps_p_eff;  // effective part of trace(eps_p)
+            lambda = 0;
+        }
+
+        using KelvinVector = ProcessLib::KelvinVectorType<DisplacementDim>;
+
+        KelvinVector eps_p_D =
+            KelvinVector::Zero();  ///< deviatoric plastic strain
+        double eps_p_V = 0;    ///< volumetric strain
+        double eps_p_eff = 0;  ///< effective plastic strain
+
+        // Initial values from previous timestep
+        KelvinVector eps_p_D_prev = KelvinVector::Zero();  ///< \copydoc eps_p_D
```

Comment:
Must be a bug in the compiler, because your suggestion works.
I think this is another reason to go for #1084 ;)